### PR TITLE
lakectl: make lakectl read environment variables 

### LINKED ISSUE #2097: Lakectl commands don't take environment variables into account
Any Lakectl command except for the "config" command not taking environment variables into account. 

Steps to reproduce: 
1. run lakefs locally 
2. run docker with the lakectl image and feed the config paraeters as environment variables
```
docker run -e LAKECTL_CREDENTIALS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE -e LAKECTL_CREDENTIALS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY -e LAKECTL_SERVER_ENDPOINT_URL=http://127.0.0.1:8000 -it treeverse/lakectl repo list 
```
3. get error: 
```
Error executing command: Get "/api/v1/repositories?after=&amount=100": unsupported protocol scheme ""
```

### PR DESCRIPTION
**Pull request by talSofer**
fixes #2097 

### Background 
lakectl uses Viper as its configuration solution. While unmarshaling configurations with Viper, the following process holds:  
1. Viper checks if each configuration key has an entry in its defaults.
2. For configurations included in the defaults, Viper searches for environment variables with that key and fetches their value if they exist. 
3. If environment variables with that key don't exist in the map, Viper takes the key's default value from defaults. 

### Problem & Root cause 
Viper never searches for environment variables for configurations that are not in the default configurations map. Therefore, lakectl does not consider environment variables.  

### Solution
For each potential configuration, add an entry to Viper's defaults with a nil value, forcing Viper to search environment variables for these configurations.

### CODE REVIEW

**Review by nopcoder — APPROVED** [review]
Suggested a way to remove the two config.go (and config package) - optional can be done on a different PR.

File: cmd/lakectl/cmd/config/config.go [hunk 1]
```diff
@@ -42,20 +46,36 @@ type configuration struct {
...
 	c.err = viper.ReadInConfig()
+	logger := logging.Default().WithField("file", viper.ConfigFileUsed())
+
+	if c.err == nil {
+		logger.Info("loaded configuration from file")
+	} else if _, ok := c.err.(viper.ConfigFileNotFoundError); !ok {
+		logger.WithError(c.err).Fatal("failed to read config file")
+	}
+
+	c.err = viper.UnmarshalExact(&c.Values)
```
Comment (nopcoder) [inline cmd/lakectl/cmd/config/config.go:75]: looks like this 'err' can shade the actual error done in the loading process.
think this one should move to the pre-command stage.

**Review by nopcoder — CHANGES_REQUESTED** [review]
added a request to change the location of the unmarshal error

File: cmd/lakectl/cmd/config/config.go [hunk 2]
```diff
@@ -64,11 +79,16 @@ const (
 	HiveDBLocationURIKey = "metastore.hive.db_location_uri"
 
 	// Defaults
-	HiveDBLocationURI = "file:/user/hive/warehouse/"
+	HiveDBLocationURI        = "file:/user/hive/warehouse/"
+	ConfigServerEndpointURL  = "server.endpoint_url"
+	DefaultServerEndpointURL = "http://127.0.0.1:8000"
+	ConfigAccessKeyID        = "credentials.access_key_id"
+	ConfigSecretAccessKey    = "credentials.secret_access_key"
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/config/config.go:86]: Note that at least the comments want to split it up into Keys and Defaults -- which doesn't happen here.  I don't particularly care, but if we don't do it then let's remove the dead comments and be consistent about the keys and defaults.

**Review by nopcoder — APPROVED** [review]
LGTM!

File: cmd/lakectl/cmd/config/config.go (hunk 2 above)
Comment (arielshaqed) [inline cmd/lakectl/cmd/config/config.go:86]: These are key names so presumably if anything gets a "Key" suffix then these should too.

Again, I don't mind either way, just please let's be consistent.

**Review by arielshaqed — APPROVED** [review]
Neat, thanks!

**Pull request merged**